source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRPresentationViewController.m

```diff
@@ -0,0 +1,44 @@
+//
+//  ACRPresentationViewController.m
+//  ADCIOS
+//
+//  Created by jwoo on 5/31/17.
+//  Copyright © 2017 Microsoft. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "ACRPresentationViewController.h"
+#import "ACRViewController.h"
+
+@implementation ACRPresentationViewController
+
+- (id) init:(NSString*) str{
+    self = [super init];
+    
+    if(self){
+        _jsonStr = nil;
+        _jsonStr = str;
+    }
+    
+    return self;
+}
+
+- (void) viewDidLoad {
+
+    [super viewDidLoad];
+    
+    if(self.jsonStr){
+        ACRViewController* acVc = [[ACRViewController alloc] init: self.jsonStr];
```

Comment:
The client code will only ever see your header files. They won't see your implementation files, because those will already be compiled into a library by the time the client gets them. That means that as long as there's no C++ in your _header_ (and there isn't any in ACRViewController.h), then the client is free to compile their own code as Objective-C using whatever compiler flags they want.

---
In reply to: [122553995](https://github.com/Microsoft/AdaptiveCards/pull/423#discussion_r122553995) [](ancestors = 122553995,122553246,122552582,122549111,122548933)